homeassistant/components/graphite/__init__.py

```diff
@@ -69,7 +69,7 @@ def setup(hass: HomeAssistant, config: ConfigType) -> bool:
     else:
         _LOGGER.debug("No connection check for UDP possible")
 
-    GraphiteFeeder(hass, host, port, protocol, prefix)
+    hass.data[DOMAIN] = GraphiteFeeder(hass, host, port, protocol, prefix)
```

Comment:
Why is this changed?

Reply:
Previously the test relied on race conditions about when the thread was done with work. Now the test just wait for the thread to be done.